docker/Dockerfile

```diff
@@ -0,0 +1,74 @@
+# StarryOS Dockerfile
+# Provides a consistent Linux build environment for StarryOS
+# This solves the issue where macOS builds fail due to hardcoded compiler names in lwext4_rust
+
+# Use a lighter base image since we install a specific Rust version anyway
+FROM debian:bookworm-slim
+
+# Install system dependencies
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    cmake \
+    clang \
+    qemu-system \
+    curl \
+    xz-utils \
+    git \
+    make \
+    dos2unix \
+    ca-certificates \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Rust using rustup
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain none
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+# Install Musl toolchain for RISC-V
+# Using musl.cc which provides pre-built cross-compilers
+RUN mkdir -p /opt/toolchains && \
+    cd /opt/toolchains && \
+    curl -f -L https://musl.cc/riscv64-linux-musl-cross.tgz -o riscv64-linux-musl-cross.tgz && \
+    tar -xzf riscv64-linux-musl-cross.tgz && \
+    rm riscv64-linux-musl-cross.tgz
+
+# Install Musl toolchain for LoongArch64 (optional, for LoongArch support)
+RUN cd /opt/toolchains && \
+    if curl -f -L https://github.com/LoongsonLab/oscomp-toolchains-for-oskernel/releases/download/loongarch64-linux-musl-cross-gcc-13.2.0/loongarch64-linux-musl-cross.tgz -o loongarch64-linux-musl-cross.tgz && \
+       tar -xzf loongarch64-linux-musl-cross.tgz && \
+       rm loongarch64-linux-musl-cross.tgz; then \
+        echo "LoongArch toolchain installed successfully."; \
+    else \
+        echo "WARNING: LoongArch toolchain download or extraction failed. LoongArch support will NOT be available."; \
+    fi
+# Set up Rust toolchain
+# Copy rust-toolchain.toml first - rustup will automatically use it
+WORKDIR /workspace
+
+# Copy rust-toolchain.toml - rustup will detect and use it
+COPY rust-toolchain.toml /workspace/rust-toolchain.toml
+
+# Install the toolchain specified in rust-toolchain.toml
+# rustup will automatically read the file when we change to the directory
+RUN cd /workspace && \
+    rustup toolchain install nightly-2025-05-20 && \
+    rustup default nightly-2025-05-20 && \
+    rustup component add rust-src llvm-tools rustfmt clippy && \
+    rustup target add x86_64-unknown-none riscv64gc-unknown-none-elf aarch64-unknown-none-softfloat loongarch64-unknown-none-softfloat
```

Comment:
Simply run `cargo -V` or some other stuffs to let rustup read `rust-toolchain.toml` content automatically.